core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -245,13 +290,30 @@ public List<TableIdentifier> listTables(Namespace namespace) {
         row ->
             JdbcUtil.stringToTableIdentifier(
                 row.getString(JdbcUtil.TABLE_NAMESPACE), row.getString(JdbcUtil.TABLE_NAME)),
-        JdbcUtil.LIST_TABLES_SQL,
+        JdbcUtil.LIST_TABLE_SQL,
         catalogName,
-        JdbcUtil.namespaceToString(namespace));
+        JdbcUtil.namespaceToString(namespace),
+        JdbcUtil.TABLE_RECORD_TYPE);
   }
 
   @Override
   public void renameTable(TableIdentifier from, TableIdentifier to) {
+    if (!tableExists(from)) {
```

Comment:
I would probably slightly change the ordering here to align with other catalog impls:
* check if namespace exists
* check if `from` exists

I think we should also align `renameView(..)` to that as well

Reply:
I have to test the table first to make `TestJdbcCatalog#testRenameTableMissingSourceTable()` happy, else I get:

```
    Expecting actual throwable to be an instance of:
      org.apache.iceberg.exceptions.NoSuchTableException
    but was:
      org.apache.iceberg.exceptions.NoSuchNamespaceException: Namespace does not exist: newdb
```

The reason is because, the test expects the `NoSuchTableException` because the `NoSuchNamespaceException` one.